src/cli/split-docs/index.ts

```diff
@@ -0,0 +1,377 @@
+import { readYaml, writeYaml } from '../../utils';
+import { red, blue, yellow, green } from 'colorette';
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+import * as mkdirp from 'mkdirp';
+import * as path from 'path';
+const isEqual = require('lodash.isequal');
+import { pathToFilename } from '../../ref-utils';
+import { isString, isObject, isNotObjectKeys } from '../../js-utils';
+import {
+  Definition,
+  Oas2Definition,
+  Oas3Definition,
+  ComponentsFiles,
+  ComponentType,
+  refObj,
+  Oas3PathItem,
+  OPENAPI3_COMPONENT,
+  COMPONENTS,
+  PATHS,
+  OPENAPI3_METHODS,
+  OPENAPI3_COMPONENTS
+} from './types'
+
+function isStartsWithComponents(node: string) {
+  const componentsPath = `#/${COMPONENTS}/`;
+  return node.startsWith(componentsPath)
+}
+
+function isNotYaml(filename: string) {
+  return !(filename.endsWith('.yaml') || filename.endsWith('.yml'));
+}
+
+function stdWriteExit(message: string) {
+  process.stderr.write(red(message));
+  process.exit(1);
+}
+
+function loadFile(fileName: string) {
+  try {
+    return yaml.safeLoad(fs.readFileSync(fileName, 'utf8')) as Definition;
+  } catch (e) {
+    return stdWriteExit(e.message);
+  }
+}
+
+function validateDefinitionFileName(fileName: string) {
+  if (!fs.existsSync(fileName)) stdWriteExit(`File ${blue(fileName)} does not exist \n`);
+  const file = loadFile(fileName);
+  if ((file as Oas2Definition).swagger) stdWriteExit('OpenAPI 2 is not supported by this command');
+  if (!(file as Oas3Definition).openapi) stdWriteExit('File does not conform to the OpenAPI Specification');
+  return true;
+}
+
+function langToExt(lang: string) {
+  const langObj: any = {
+    php: '.php',
+    'c#': '.cs',
+    shell: '.sh',
+    curl: '.sh',
+    bash: '.sh',
+    javascript: '.js',
+    js: '.js',
+    python: '.py'
+  }
+  return langObj[lang];
+}
+
+function traverseFolderDeep(folder: string, callback: any) {
```

Comment:
We are mixing between the word folder and directory. Be consistent and use one. Let’s use directory.